src/detect-content.c

```diff
@@ -407,21 +407,22 @@ bool DetectContentPMATCHValidateCallback(const Signature *s)
         if (sm->type != DETECT_CONTENT)
             continue;
         const DetectContentData *cd = (const DetectContentData *)sm->ctx;
-        uint32_t right_edge = cd->content_len + cd->offset;
         if (cd->content_len > max_right_edge) {
             SCLogError(SC_ERR_INVALID_SIGNATURE,
                     "signature can't match as content length %u is bigger than dsize %u.",
                     cd->content_len, max_right_edge);
-            return FALSE;
+            return false;
         }
+        /* Incorporate the distance when comparing with the right edge */
+        uint32_t right_edge = cd->content_len + cd->offset + cd->distance;
```

Comment:
If there is  distance keyword, it means there is a previous content keyword as well. Ideally we'd also take this into account. I think ultimately we'd need logic to take all the relevant keywords and consider them all. But this would be quite complex.

Minor note here: distance is signed and can be negative. It can be used to look before the current pattern I think, so it might lead to underflow of `right_edge`